### PR TITLE
classify user question into op-forum topic categories

### PR DESCRIPTION
**Pull request by devjoaov**
image
image

### CODE REVIEW

File: pkg/op-app/op_app/api.py [hunk 1]
```diff
@@ -73,12 +74,22 @@ def handle_exception(e):
...
-@handle_question
-async def predict(question, memory):
-    result = await process_question(question, memory)
+async def capture_question_classification(question, memory, user_token):
+    result = await classify_question(question, memory)
+    classification = result["classification"]
+
+    posthog.capture(
+        user_token,
+        "QUESTION_CLASSIFICATION",
```
Comment (ribeirojose) [inline pkg/op-app/op_app/api.py:83]: Why emit a new event? Couldn't we just put this in the event that already exists?

File: pkg/op-brains/op_brains/chat/classify.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from langchain_openai import ChatOpenAI
+from langchain_anthropic import ChatAnthropic
+from typing import List, Tuple
+from op_brains.summarizer.utils import Prompt, QuestionClassificationStructured
+from op_brains.chat.apis import access_APIs
+
+
+async def classify_question(question: str, thread: List = []) -> str:
```
Comment (ribeirojose) [inline pkg/op-brains/op_brains/chat/classify.py:8]: I thought the approach was to use the categories from the forum using the references used in the response. Can you explain more about why you ended up using this approach? This way we cant ever guarantee there's only going to be a limited set of categories right? How did you think we'd analyze this data?

Also, doesn't this add even more latency to the answer? Do you think the value added is worth it compared to the other approach?

**Review by ribeirojose — APPROVED** [review]
Looks ok to me, but left a question which might be relevant

File: pkg/op-app/op_app/api.py (hunk 1 above)
Comment (devjoaov) [reply pkg/op-app/op_app/api.py:83]: we can do it 👍

**Review by ribeirojose — APPROVED** [review]
👌 perfect, thanks!

**Pull request merged**